elastalert/elastalert.py

```diff
@@ -675,8 +682,8 @@ def alert(self, matches, rule, alert_time=None):
                     qk = match[rule['query_key']]
                 else:
                     qk = None
-                start = ts_add(match[rule['timestamp_field']], -rule.get('timeframe', datetime.timedelta(minutes=10)))
-                end = ts_add(match[rule['timestamp_field']], datetime.timedelta(minutes=10))
+                start = ts_to_dt(match[rule['timestamp_field']]) - rule.get('timeframe', datetime.timedelta(minutes=10))
+                end = ts_to_dt(match[rule['timestamp_field']]) + datetime.timedelta(minutes=10)
```

Comment:
Extremely pedantic: at DST cutovers, you have a race condition that might widen or shorten the window if DST switches between when start and end are calculated.

It's unlikely to ever appear, but you should consider storing a current time variable to prevent this in future code.